libtock/touch.h

```diff
@@ -0,0 +1,41 @@
+// Touch Panel Library
+
+#pragma once
+
+#include "tock.h"
+
+#define DRIVER_NUM_TOUCH 0x90002
+
+#define TOUCH_STATUS_RELEASED 0
+#define TOUCH_STATUS_PRESSED 1
+#define TOUCH_STATUS_MOVED 2
+
+#define GESTURE_NO 0
+#define GESTURE_SWIPE_UP 1
+#define GESTURE_SWIPE_DOWN 2
+#define GESTURE_SWIPE_LEFT 3
+#define GESTURE_SWIPE_RIGHT 4
+#define GESTURE_ZOOM_IN 5
+#define GESTURE_ZOOM_OUT 6
+
+typedef void (touch_callback)(int, int, int, void*);
+typedef void (gesture_callback)(int, void*);
+
+// buffer data format
+//  0         1           2                  4                  6           7             8         ...
+// +---------+-----------+------------------+------------------+-----------+---------------+--------- ...
```

Comment:
Can you define a structure with this format? Hardcoded offsets are buggy.

Reply:
Your suggestion would be to use a packed structure instead of the buffer?